File: rBike.Services/Database/Report.cs
using System;
using System.Collections.Generic;

namespace rBike.Services.Database;

public partial class Report
{
    public int ReportId { get; set; }

    public int Month { get; set; }

    public int Year { get; set; }

    public decimal TotalValue { get; set; }

    public string TopProductsJson { get; set; } = null!;

    public DateTime CreatedAt { get; set; }

    public int CreatedById { get; set; }

    public string PdfFilePath { get; set; } = null!;

    public virtual User CreatedBy { get; set; } = null!;
}

File: UI/rbike_admin/lib/models/report.dart
class Report {
  final int reportId;
  final int month;
  final int year;
  final double totalValue;
  final String topProductsJson;
  final DateTime createdAt;
  final int createdById;
  final String pdfFilePath;

  Report({
    required this.reportId,
    required this.month,
    required this.year,
    required this.totalValue,
    required this.topProductsJson,
    required this.createdAt,
    required this.createdById,
    required this.pdfFilePath,
  });

  factory Report.fromJson(Map<String, dynamic> json) {
    return Report(
      reportId: json['reportId'],
      month: json['month'],
      year: json['year'],
      totalValue: (json['totalValue'] as num).toDouble(),
      topProductsJson: json['topProductsJson'],
      createdAt: DateTime.parse(json['createdAt']),
      createdById: json['createdById'],
      pdfFilePath: json['pdfFilePath'],
    );
  }
} 